credential.c

```diff
@@ -301,6 +302,14 @@ static int run_credential_helper(struct credential *c,
 {
 	struct child_process helper = CHILD_PROCESS_INIT;
 	FILE *fp;
+	struct timeval select_timeout = {
+		.tv_sec = 5 * 60, /* Five minutes. */
+	};
+	int config_seconds;
+
+	if (!git_config_get_int("credential.timeout", &config_seconds) &&
+	    config_seconds >= 0)
+		select_timeout.tv_sec = config_seconds;
```

Comment:
Am I reading this right?  If `credential.timeout` is not defined in the config, we'll default to 5 minutes?  So to avoid the timeout and get the legacy wait, we need to have `credential.timeout=0` in the config.  Not questioning, just confirming intent.

Reply:
Yes, that's the intent.

Reply:
Just a note for GCM.. this is a setting we should ignore ourselves :)